Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -584,6 +584,12 @@ def audit_homepage
       problem "Please use https:// for #{homepage}"
     end
 
+    case homepage
+    when %r{^http://([^/]*)sourceforge\.net(/|$)}i,
+         %r{^http://([^/]*)sf\.net(/|$)}i
+      problem "#{homepage} should be `https://#{$1}sourceforge.io/`"
```

Comment:
Regarding the other existing [checks](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/dev-cmd/audit.rb#L553:L560), you could use the following problem message: 

`problem "Please use https:// for #{homepage}"`

to simply extend the existing [tests](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/audit_test.rb#L426:L434) with:

```ruby
    formula_homepages = {
      # other sites
      "projectlibre" => "http://sourceforge.net/projects/projectlibre/",
      "sevenzup" => "http://sf.net/projects/sevenzip/"     
    }
```